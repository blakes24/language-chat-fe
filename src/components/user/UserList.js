import Container from "@material-ui/core/Container";
import { makeStyles } from "@material-ui/core/styles";
import UserCard from "./UserCard";

const useStyles = makeStyles({
  root: {
    flexGrow: 1,
    padding: 0,
    margin: "0 auto",
    maxWidth: "700px",
  },
});

function UserList({ users }) {
  const classes = useStyles();
  return (
    <Container className={classes.root}>
      {users.map((user) => (
        <UserCard cardUser={user} key={user.id} />
      ))}
    </Container>
  );
}

export default UserList;
