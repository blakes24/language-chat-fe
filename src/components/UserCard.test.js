import { render } from "@testing-library/react";
import UserList from "./UserList";

const users = [
  {
    id: 1,
    name: "Brady",
    bio: "Eos nobis autem sint ducimus illo.",
    imageUrl: "https://cdn.fakercloud.com/avatars/happypeter1983_128.jpg",
    active: true,
    speaks: [
      {
        code: "it",
        language: "Italian",
      },
    ],
    learning: [
      {
        code: "ko",
        language: "Korean",
        level: "2",
      },
    ],
  },
];

test("renders without crashing", () => {
  render(<UserList users={users} />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<UserList users={users} />);
  expect(asFragment()).toMatchSnapshot();
});
