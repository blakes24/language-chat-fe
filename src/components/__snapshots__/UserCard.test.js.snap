// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root makeStyles-root-12 MuiContainer-maxWidthLg"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-root-13 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <div
          class="makeStyles-top-16"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded makeStyles-avatar-15"
          >
            <img
              alt="Brady"
              class="MuiAvatar-img"
              src="https://cdn.fakercloud.com/avatars/happypeter1983_128.jpg"
            />
          </div>
          <div
            class="makeStyles-details-18"
          >
            <div
              class="makeStyles-items-17"
            >
              <h2
                class="MuiTypography-root MuiTypography-h6"
              >
                Brady 
              </h2>
              <div>
                <button
                  aria-label="follow"
                  class="MuiButtonBase-root MuiIconButton-root makeStyles-btn-22 MuiIconButton-colorSecondary"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm-9-2V7H4v3H1v2h3v3h2v-3h3v-2H6zm9 4c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="send a message"
                  class="MuiButtonBase-root MuiIconButton-root makeStyles-btn-22 MuiIconButton-colorSecondary"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM6 9h12v2H6V9zm8 5H6v-2h8v2zm4-6H6V6h12v2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <div
              class="makeStyles-langs-20"
            >
              <div
                class="makeStyles-langSpace-21"
              >
                Italian 
                <div
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuenow="100"
                  class="MuiLinearProgress-root MuiLinearProgress-colorPrimary makeStyles-level-19 MuiLinearProgress-determinate"
                  role="progressbar"
                >
                  <div
                    class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
                    style="transform: translateX(0%);"
                  />
                </div>
              </div>
              <div>
                Korean 
                <div
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuenow="50"
                  class="MuiLinearProgress-root MuiLinearProgress-colorPrimary makeStyles-level-19 MuiLinearProgress-determinate"
                  role="progressbar"
                >
                  <div
                    class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
                    style="transform: translateX(-50%);"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <p
          class="MuiTypography-root makeStyles-bio-14 MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          Eos nobis autem sint ducimus illo.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
